### PR TITLE
[idea] prevent (follow-up) NPEs

### PR DESCRIPTION
**Pull request by svenefftinge**

### CODE REVIEW

**Comment by svenefftinge** [comment]
Jenkins intellij-tests-master-latest

File: intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/build/BuildProgressReporter.xtend [hunk 1]
```diff
@@ -57,7 +57,7 @@ class BuildProgressReporter implements BuildRequest.IPostValidationCallback {
 
 	override afterValidate(URI validated, Iterable<Issue> issues) {
 		markAsAffected(validated)
-		for (issue : issues) {
+		for (issue : issues.filterNull) {
```
Comment (szarnekow) [inline intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/build/BuildProgressReporter.xtend]: Would it be better if we prevent null-issues in the iterable in the first place?

Comment (svenefftinge) [inline intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/build/BuildProgressReporter.xtend]: The iterable comes from a language implementation, so we cannot guarantee it. The default impl, shouldn't do that, though.

Comment (szarnekow) [inline intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/build/BuildProgressReporter.xtend]: I see - I got the impression that you had an NPE and added the guard afterwards here.

Comment (svenefftinge) [inline intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/build/BuildProgressReporter.xtend]: Yes, I had one. I wanted to say that we cannot generally make sure that IResourceValidators are implemented correctly. So this null guard should be there.

**Pull request merged**